### PR TITLE
BFD-531: Removed the jessentials library

### PR DESCRIPTION
**Pull request by karlmdavis**
### Change Details

Removed the jessentials library. Replaced the exception classes it was providing with a combo of newer JDK classes and compatible re-creations of the original ones, placed in the `bfd-shared-utils` module.

### Acceptance Validation

Acceptance criteria:

1. Does everything still work?

### Feedback Requested

Please let me know:

1. When you close your eyes and think of this PR, what color do you see?

### External References

- [BLUEBUTTON-531](https://jira.cms.gov/browse/BLUEBUTTON-531)

### Security Implications

No security implications noted for this change set:

- [ ] new software dependencies

- [ ] altered security controls

- [ ] new data stored or transmitted

- [ ] security checklist is completed for this change

- [ ] requires more information or team discussion to evaluate security implications

### CODE REVIEW

File: apps/bfd-server/bfd-server-war/pom.xml [hunk 1]
```diff
@@ -115,6 +115,13 @@
 			</exclusions>
 		</dependency>
 
+		<dependency>
```
Comment (karlmdavis) [inline apps/bfd-server/bfd-server-war/pom.xml:118]: Please note that BFD already had this dependency: it was getting it transitively from `jessentials`. Now that that's gone, we're just including it directly. No SIA should be needed.

**Review by jzulim — APPROVED** [review]
Thank you for doing this. LGTM. When I close my eyes I see 'Black'

**Review by cbrunefearless — APPROVED** [review]
lgtm

**Review by cbrunefearless — APPROVED** [review]
lgtm

**Review by jzulim — APPROVED** [review]
Looks good, should make maintenance much better. When I close my eyes I see 'Black and Brown static"

**Comment by karlmdavis** [comment]
> Looks good, should make maintenance much better. When I close my eyes I see 'Black and Brown static"

@jzulim https://theswaddle.com/seeing-colors-when-eyes-closed-phosphenes/

**Pull request merged**